lib/MumbleConnection.js

```diff
@@ -554,8 +611,15 @@ MumbleConnection.prototype._processAudio = function() {
             // We got listeners for voice event so do decode.
             var decoded = [];
             for( var f in packets ) {
+                var data;
+                if(this.opusEnabled) {
+                    data = packets[f].decoder.decode( packets[f].frame, this.FRAME_SIZE );
+                }
+                else {
+                    data = packets[f].decoder.decode( packets[f].frame );
+                }
                 decoded.push({
-                    data: packets[f].decoder.decode( packets[f].frame ),
+                    data: data,
                     codec: packets[f].codec,
```

Comment:
Yeah. But it's not the server switching what matters though. :)

There's latency involved in the network communications:

     Client 1          Server          Client 2
    
    OPUS packet
                 \
                  \
                   \_____________
                                 \
    OPUS packet                   \
                 \                 \
                  \                   OPUS received
                   \_____________
                                 \
                    Server sends  \
                    CodecVersion   \
                    _____________     OPUS received 
    OPUS packet    /             \
                 \/               \
                 /\                \
    CELT switch    \_____________     CELT switch
                                 \
                Server does NOT   \
             re-encode the packet  \
                                      OPUS received
    

Not to mention the two packets should be sent over different transports (Currently Node-mumble doesn't support UDP for audio, but hopefully in future we do - and while waiting for that the native Mumble clients support it already).

This gives us the following scenario:

     Client 1                           Server           Client 2
    
    CELT packet  ------------------>   Broadcast  ---> Receive CELT
    CELT packet  ------------------>   Broadcast  ---> Receive CELT
    CELT packet  ------------------>   Broadcast  ---> Receive CELT
                      (Lost) <------ Client leave ---> OK
    CELT packet  ------------------>   Broadcast  ---> Receive CELT
     (Buffered)  <------------------ CodecVersion ---> Change to OPUS
    CELT packet  ------------------>   Broadcast  ---> Receive CELT
                                  ...
    CELT packet  ------------------>   Broadcast  ---> Receive CELT
             OK  <-(Retransmission)- Client leave
             OK  <--(From buffer)--- CodecVersion     
    OPUS packet  ------------------>   Broadcast  ---> Receive OPUS

Given a one-way latency of 50ms between the server and the client, that'll mean the client will send 50ms worth of audio with the wrong codec before it receives the order to change codec. If the order to change codec is delayed because of a lost packet, the retransmission won't occur until 2x round-trip time (2x100ms), which will cause further 200ms worth of audio with the wrong codec.